src/server/json_family.cc

```diff
@@ -905,8 +933,73 @@ OpResult<vector<OptString>> OpMGet(const OpArgs& op_args, const vector<string_vi
   return vec;
 }
 
+// Returns numeric vector that represents the number of fields of JSON value at each path.
+OpResult<vector<OptSizeT>> OpFields(const OpArgs& op_args, string_view key,
+                                    JsonExpression expression) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  vector<OptSizeT> vec;
+  auto cb = [&vec](const string_view& path, const json& val) {
+    vec.emplace_back(CountJsonFields(val));
+  };
+
+  expression.evaluate(*result, cb);
+  return vec;
+}
+
 }  // namespace
 
+void JsonFamily::Debug(CmdArgList args, ConnectionContext* cntx) {
+  function<decltype(OpFields)> func;
```

Comment:
I am not sure I understand why you need this "func" complication.
why not just call `OpFields` at line 990?